### PR TITLE
Remove authEnabled flag

### PR DESCRIPTION
**Pull request by matt-royal**
## Is there a related GitHub Issue?

#445

## What is this change about?

See issue

## Does this PR introduce a breaking change?

No

## Acceptance Steps

See issue

## Tag your pair, your PM, and/or team

@akrishna90 

## Things to remember

#898 will need to be merged also

### CODE REVIEW

**Review by gnovv — APPROVED** [review]
lgtm

File: go.mod [hunk 1]
```diff
@@ -38,6 +38,8 @@ require (
 	sigs.k8s.io/hierarchical-namespaces v0.9.0
 )
 
+require github.com/stretchr/testify v1.7.1 // indirect
```
Comment (gcapizzi) [inline go.mod:41]: wondering why this is happening 🤔  it should have gone in the block below with all the other indirect dependencies, right?

**Review by gcapizzi — APPROVED** [review]

**Comment by kieron-dev** [comment]
Please ignore the `Concourse / e2e-tests` failure. The PR requires the correct kbld config from commit 4ae4a93 to be able to deploy via concourse, but it's not worth rebasing just for that.

File: go.mod (hunk 1 above)
Comment (matt-royal) [reply go.mod:41]: We noticed that, too. I think that `go mod tidy` added it, but I'm not sure why.

**Pull request merged**